File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EBackgroundDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EBackgroundDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd1
            AppTheme.STANDARD -> R.color.cstd1
            AppTheme.BLUE -> R.color.cbd1
            AppTheme.GREEN -> R.color.cgd1
            AppTheme.RED -> R.color.crd1
            AppTheme.NOTHING -> R.color.cnod1
        }
    }
}

File: lib/layer2_core/dark/ebackground_dark.dart
import 'package:flutter/material.dart';

// Import RAW Dark Values
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

import '../../layer3_logic/design_engine_controller.dart'; // Für AppTheme enum

/// Layer 2: Core Tokens - eBackground (Dark Mode)
/// Mappt das gewählte Theme auf die entsprechende Konstante aus Schicht 1.
class EBackgroundDark {
  static Color getColor(AppTheme activeTheme) {
    switch (activeTheme) {
      case AppTheme.system:
        return csyd1;
      case AppTheme.standard:
        return cstd1;
      case AppTheme.blue:
        return cbd1;
      case AppTheme.green:
        return cgd1;
      case AppTheme.red:
        return crd1;
      case AppTheme.nothing:
        return cnod1;
    }
  }
}
